### PR TITLE
Introduce NodeDag, a wrapper around a networkx graph

### PR DESCRIPTION
**Pull request by turbo1912**

### CODE REVIEW

File: src/fal/node_graph.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        self.nodeLookup: Dict[str, FalFlowNode] = {}
+
+        for model in falDbt.list_models():
+            model_fal_node = DbtModelNode(model.unique_id, model)
+            self.nodeLookup[model_fal_node.unique_id] = model_fal_node
+            self.graph.add_node(model_fal_node.unique_id)
+
+            # Add dbt model dependencies
+            for dbt_dependency_unique_id in model_fal_node.model.node.depends_on_nodes:
+                self.graph.add_node(dbt_dependency_unique_id)
```
Comment (chamini2) [inline src/fal/node_graph.py:48]: I am guessing this is idempotent?

File: src/fal/node_graph.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    model: DbtModel
+    pass
+
+
+@dataclass
+class NodeGraph:
+    "Wrapper around networkx graph"
+
+    def __init__(self, falDbt: FalDbt):
+        self.falDbt = falDbt
```
Comment (chamini2) [inline src/fal/node_graph.py:37]: rename to `fal_dbt`?

File: src/fal/node_graph.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            self.nodeLookup[fal_script_node.unique_id] = fal_script_node
+            # fal_script_node depends on model_fal_node
+            self.graph.add_edge(
+                fal_script_node.unique_id, downstream_fal_node_unique_id
+            )
+
+
+def _script_id_from_path(scriptPath: Path, modelName: str):
+    script_file_name = os.path.basename(scriptPath)
+    return f"script.{modelName}.{script_file_name}"
```
Comment (chamini2) [inline src/fal/node_graph.py:101]: Good solution

**Review by chamini2 — APPROVED** [review]
Good! Some tests would be great, as you mentioned

File: src/fal/node_graph.py (hunk 1 above)
Comment (turbo1912) [reply src/fal/node_graph.py:48]: yup

**Pull request merged**